### PR TITLE
switch to hyperkube and remove renames

### PR DESCRIPTION
**Pull request by deads2k**
This publishes hyperkube and switches to use it instead of renaming the openshift binary.

### CODE REVIEW

**Comment by deads2k** [comment]
wow, that worked better than it had any right to.

File: test/cmd/help.sh [hunk 1]
```diff
@@ -13,10 +13,7 @@ os::cmd::expect_success 'openshift admin config'
 os::cmd::expect_success 'openshift cli config'
 os::cmd::expect_success 'openshift admin policy '
 os::cmd::expect_success 'openshift kubectl'
-os::cmd::expect_success 'openshift kube'
 os::cmd::expect_success 'openshift admin'
-os::cmd::expect_success 'openshift start kubernetes'
-os::cmd::expect_success 'kubernetes'
```
Comment (mfojtik) [inline test/cmd/help.sh]: can you add test for hyperkube?

**Comment by mfojtik** [comment]
LGTM

File: origin.spec [hunk 2]
```diff
@@ -421,12 +415,7 @@ install -p -m 755 contrib/migration/* %{buildroot}%{_datadir}/%{name}/migration/
 %doc README.md
 %license LICENSE
 %{_bindir}/openshift
-%{_bindir}/kube-apiserver
```
Comment (stevekuznetsov) [inline origin.spec]: Do we need symlinks here for backwards compatibility?

Comment (deads2k) [reply origin.spec]: > Do we need symlinks here for backwards compatibility?

A symlink wouldn't work.  You could create a bash script if you really wanted to.  @smarterclayton didn't indicate it was a requirement.

Comment (stevekuznetsov) [reply origin.spec]: /cc @sdodson

Comment (sdodson) [reply origin.spec]: We don't have any dependencies on kube commands in the installer. This should be fine.

**Comment by smarterclayton** [comment]
You're now building hyperkube twice:

```
++ Building go targets for linux/amd64: images/pod examples/hello-openshift
++ Building go targets for linux/amd64: vendor/k8s.io/kubernetes/cmd/hyperkube pkg/network/sdn-cni-plugin vendor/github.com/containernetworking/cni/plugins/ipam/host-local vendor/github.com/containernetworking/cni/plugins/main/loopback
++ Building go targets for linux/amd64: cmd/openshift cmd/oc cmd/kubefed cmd/template-service-broker vendor/k8s.io/kubernetes/cmd/hyperkube
++ Building go targets for linux/amd64: test/extended/extended.test
```

Need to remove it from the IMAGE specific targets now that you made it a normal target.

**Comment by deads2k** [comment]
> Need to remove it from the IMAGE specific targets now that you made it a normal target.

Image targets?  What have we built?!

**Comment by smarterclayton** [comment]
:)

Image targets are things we only build for Linux today and which are
primarily used in images.  The stock cross compile target is for anything
we expect to work across all platforms for dev and test and end users.

I guess we’ll want hyperkube in cross at some point, but only when we
actually do the split.

On Nov 22, 2017, at 7:32 AM, David Eads <notifications@github.com> wrote:

Need to remove it from the IMAGE specific targets now that you made it a
normal target.

Image targets? What have we built?!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/17369#issuecomment-346337150>,
or mute
the thread

**Pull request merged**